docs: refactor of documentation (#2)
SmartWasteCollection/documentation: requirements/use-cases.md

# Use Cases

> The following use cases diagrams describes the behavior of the system.

Reviewer: credo fosse giusto prima perchè use cases è plurale

> The following use cases diagrams describes the behavior of the system.

Reviewer 2: ```suggestion
The following use cases diagrams describe the behavior of the system.
```

> 

Reviewer: Metto il commento qui ma in realtà mi riferisco alla riga sopra che non posso commentare perchè non è stata modificata. Abbiamo riusato la stessa descrizione dell' Ordinary Waste Collection ma in questo caso i Truck non sono coinvolti quindi proporrei di modificarla così:
```suggestion
Regarding ordinary waste disposal, the main actors are citizens and dumpsters. 
```

<!--![Diagram Image Link](./ordinary-disposal-use-cases.pm.puml)-->

Citizens can dispose of waste inside dumpsters only if they use their smart card to make the opening request. Once the citizen has made the deposit, the dumpster will update its occupied volume and perform the automatic closure, which will occur after a predetermined amount of time.

### Ordinary Waste Collection

Regarding ordinary waste collection, the main actors are trucks and dumpsters. In this scenario, trucks and truck drivers play the same role.

<!--![Diagram Image Link](./ordinary-collection-use-cases.pm.puml)-->

During its mission, a garbage truck shares its current location with the system and updates the mission status.

> At each step of the mission, the truck will update its current occupied volume and the emptied bin will do the same as well.

Reviewer 2: ```suggestion
At each step of the mission, the truck will update its current occupied volume and the emptied dumpster will do the same.
```

> A mission is generated because at least one dumpster of the system has made a _full dumpster_ notification.

Reviewer 2: ```suggestion
A mission is generated because at least one dumpster has sent a _full dumpster_ notification.
```
